wikimedia/wikipedia-ios
# Perform deeper database cleanup upon Settings clear cache

**Phabricator:** 
https://phabricator.wikimedia.org/T341104

### Notes
This PR adds some additional database and caching cleanup logic to the Settings > Clear Cache action. It leans on our existing `WMFDatabaseHousekeeper` and  `SharedContainerCacheHousekeeping` (which we currently trigger via BackgroundTasks). I'm passing in a new `WMFCleanupLevel` parameter to indicate that from these contexts we want to do a deeper clean in hopes of removing vandalism. In regards to the requirements of https://phabricator.wikimedia.org/T341104, we are deleting all Explore feed data and refreshing it upon completion.

### Test Steps
1. Load several days worth of Explore feed data.
2. Tap Clear Cache via Settings.
3. Main thread should **not** be blocked, and an alert should appear when the cleanup task has completed.
4. Explore feed should refresh.
